File: GroupAnagrams_49.cpp
//Given an array of strings strs, group the anagrams together. You can return the answer in any order.

#include <vector>
#include <unordered_map>

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        std::unordered_map<string, int> groupMap;
        std::vector<std::vector<string>> outList;

        int c = 0;
        for (int i = 0; i < strs.size(); ++i){
            std::string s = strs[i];
            std::sort(s.begin(), s.end());

            if (groupMap.count(s) < 1){
                groupMap[s] = c;
                std::vector<string> newVec = {strs[i]};
                outList.push_back(newVec);
                c++;
            }

            else{
                int n = groupMap[s];
                outList[n].push_back(strs[i]);
            }
        }

    return outList;
    }
};

File: GroupAnagrams_49.py
#Given an array of strings strs, group the anagrams together. You can return the answer in any order.

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        group_map = dict()
        outlist = []
        
        i = 0
        for str in strs:
            s = ''.join(sorted(str))

            if s not in group_map:
                group_map[s] = i 
                outlist.append([str])
                i += 1
            else:
                n = group_map[s]
                outlist[n].append(str)
                
        return outlist